test/integration/master/kms_transformation_test.go

```diff
@@ -140,22 +122,67 @@ func TestKMSProvider(t *testing.T) {
 	}
 
 	// Secrets should be un-enveloped on direct reads from Kube API Server.
-	s, err := test.restClient.CoreV1().Secrets(testNamespace).Get(testSecret, metav1.GetOptions{})
+	mustReadSecret(t, test, true)
+	test.printMetrics()
+}
+
+func mustSetupTest(t *testing.T) (*base64Plugin, *transformTest) {
+	pluginMock, err := NewBase64Plugin()
+	if err != nil {
+		t.Fatalf("failed to create mock of KMS Plugin: %v", err)
+	}
+
+	test, err := newTransformTest(t, kmsConfigYAML)
+	if err != nil {
+		pluginMock.cleanUp()
+		t.Fatalf("failed to start KUBE API Server with encryptionConfig\n %s", kmsConfigYAML)
+	}
+
+	// As part of newTransformTest a new secret was created, so KMS Mock should have been exercised by this point.
+	if len(pluginMock.errorChan) != 0 {
+		pluginMock.cleanUp()
+		test.cleanUp()
+		t.Fatalf("KMSPlugin failed while serving requests: %v", <-pluginMock.errorChan)
+	}
+
+	// Ensuring that the test secret has been cached - written to etcd and read back by cacher.
+	pollErr := wait.PollImmediate(1*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+		_, err := test.restClient.CoreV1().Secrets(test.ns.Name).Get(test.secret.Name, metav1.GetOptions{ResourceVersion: "0"})
+		return err == nil, nil
```

Comment:
If you get an error that isn't "Not Found", the test will fail, but the output will say that it "timed out".
Is there a way to check the kind of err here and return (false, err) if it's something unexpected?

If that's tricky, maybe just log the err every time